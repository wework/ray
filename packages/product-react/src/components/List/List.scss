@import url('https://www-static.wework.com/apercu/apercu.css');
@import url('https://www-static.wework.com/apercu/apercu_mono.css');

$list-font-family: 'Apercu', 'Apercu-Regular', 'Apercu', Avenir, -apple-system,
  BlinkMacSystemFont, 'Helvetica Neue', Helvetica, 'Calibri', 'Roboto', Arial,
  sans-seri;
$list-font-weight: normal;
$list-block-display: block;
$list-inline-block-display: inline-block;
$list-vertical-align: middle;
$list-border-radius: 4px;
$list-icon-block-margin: 12px 16px 12px 12px;

.ray-list-item {
  font-family: $list-font-family;
  font-weight: $list-font-weight;
  $list-border-radius: 4px;
  background-color: rgb(247, 247, 247);

  &__header {
    color: rgb(0, 0, 0);
    font-family: $list-font-family;
    font-size: 16px;
    font-weight: $list-font-weight;
    display: $list-block-display;
    vertical-align: $list-vertical-align;
    line-height: 20px;
  }

  &__content {
    color: rgb(102, 102, 102);
    font-family: $list-font-family;
    font-size: 12px;
    font-weight: $list-font-weight;
    height: 16px;
    display: $list-block-display;
    vertical-align: $list-vertical-align;
    line-height: 16px;
  }

  &__container {
    display: $list-inline-block-display;
    vertical-align: $list-vertical-align;
    margin-top: 5px;
    margin-bottom: 5px;
    line-height: 1;
  }

  &__icon-block {
    display: $list-inline-block-display;
    vertical-align: $list-vertical-align;
    line-height: 1;
    margin: $list-icon-block-margin;
  }
}
