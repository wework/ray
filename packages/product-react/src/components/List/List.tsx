import clsx from 'clsx';
import React from 'react';
import { Avatar } from '../Avatar';
import './List.scss';

export interface IItems {
  header?: string;
  content?: string;
  icon?: string;
}

export interface IListProps extends React.HTMLProps<HTMLLIElement> {
  items: IItems[];
  listItemComponent?: any;
  showAvatar?: boolean;
  onClick?(event: React.MouseEvent<HTMLLIElement>): void;
}

const ListItem: React.FC<IListProps> = ({
  items,
  showAvatar,
  className,
  ...rest
}) => {
  let listItems: React.ReactNode[];
  listItems = items.map((item: IItems, index: number) => {
    return (
      <li className={className} {...rest} key={index}>
        {showAvatar && (
          <Avatar
            className="ray-list-item__icon-block"
            image={item.icon}
            size="medium"
            name={item.header}
          />
        )}
        <p className="ray-list-item__container">
          <span className="ray-list-item__header">{item.header}</span>
          <span className="ray-list-item__content">{item.content}</span>
        </p>
      </li>
    );
  });
  return <ul>{listItems}</ul>;
};

export const List: React.FC<IListProps> = ({
  items,
  selected,
  listItemComponent,
  className,
  ...rest
}) => {
  const Component = listItemComponent;
  const wrapperClass = clsx('ray-list-item', className);

  return <Component items={items} className={wrapperClass} {...rest} />;
};

List.defaultProps = {
  items: [],
  listItemComponent: ListItem
};

export default List;
