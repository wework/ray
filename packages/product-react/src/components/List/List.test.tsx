import Enzyme, { configure, mount, ReactWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { List } from './List';

configure({ adapter: new Adapter() });

describe('List', () => {
  let component: ReactWrapper<any>;

  beforeEach(() => {
    component = mount(<List items={[{}]} />);
  });

  test('should match snapshot and styles', () => {
    expect(
      mount(
        <List
          showAvatar={true}
          items={[{ header: 'header', content: 'content', icon: '' }]}
        />
      )
    ).toMatchSnapshot();
  });
});
