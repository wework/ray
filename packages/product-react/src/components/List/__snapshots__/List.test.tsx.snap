// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot and styles 1`] = `
<List
  items={
    Array [
      Object {
        "content": "content",
        "header": "header",
        "icon": "",
      },
    ]
  }
  listItemComponent={[Function]}
  showAvatar={true}
>
  <ListItem
    className="ray-list-item"
    items={
      Array [
        Object {
          "content": "content",
          "header": "header",
          "icon": "",
        },
      ]
    }
    showAvatar={true}
  >
    <ul>
      <li
        className="ray-list-item"
        key="0"
      >
        <Avatar
          add={false}
          className="ray-list-item__icon-block"
          edit={false}
          image=""
          name="header"
          size="medium"
        >
          <div
            className="lucia-avatar lucia-avatar--medium ray-list-item__icon-block"
          >
            <span
              className="ray-text--body"
            >
              h
               
            </span>
          </div>
        </Avatar>
        <p
          className="ray-list-item__container"
        >
          <span
            className="ray-list-item__header"
          >
            header
          </span>
          <span
            className="ray-list-item__content"
          >
            content
          </span>
        </p>
      </li>
    </ul>
  </ListItem>
</List>
`;
